src/expr.rs

```diff
@@ -320,6 +320,19 @@ fn rewrite_closure(capture: ast::CaptureClause,
     Some(format!("{} {}", prefix, try_opt!(body_rewrite)))
 }
 
+fn nop_block_collapse(bl_str_o: Option<String>) -> Option<String> {
+    if let Some(ref bl_str) = bl_str_o {
```

Comment:
I think you can write this function more elegantly as `bl_str_o.map(|bl_str| { ... })`, with a slight adjustment to the body